### PR TITLE
New blog: Snowflake Catalog

### PR DESCRIPTION
**Pull request by scottteal**
Adding new blog post announcing Snowflake Catalog in Iceberg 1.2.0. Please review

### CODE REVIEW

**Review by nastra — APPROVED** [review]
LGTM once merge conflicts have been resolved

**Review by Fokko — APPROVED** [review]
Thanks for adding this blog @scottteal, for some reason I missed this one. Thanks @nastra for the review

**Pull request merged**
